File: services/main/packages/graphql/model/partlist.go
package model

import (
	"wrs/tk/packages/core/partlist"
)

type PartList struct {
	ID        int64
	Name      string
	Parent_ID int64
}

func ToPartList(p *partlist.PartList) PartList {
	ret := PartList{
		ID:        p.ID,
		Name:      p.Name,
		Parent_ID: p.Parent_ID.Int64,
	}

	return ret
}

File: services/nginx/frontend/src/stores/partlist.ts
import { defineStore } from "pinia"
import { Ref, ref } from "vue"

// ## Graphql Schema Definition
// type PartList {
//   id: Int64!
//   name: String!
//   parent_id: Int64
// }

export type PartList = {
  id: number
  name: string
  parent_id: number
  children: PartList[]
}

export const usePartListStore = defineStore("partlist", () => {
  const partLists: Ref<PartList[]> = ref([])
  function setPartLists(list: PartList[]) {
    partLists.value = list
  }
  function deletePartList(id: number) {
    partLists.value = partLists.value.filter((value) => {
      return value.id !== id
    })
  }
  function addPartList(newList: PartList) {
    partLists.value.push(newList)
  }
  return {
    partLists,
    setPartLists,
    deletePartList,
    addPartList,
  }
})
